libethereum/Account.h

```diff
@@ -152,29 +140,24 @@ class Account
 	/// database.
 	void setStorageCache(u256 _p, u256 _v) const { const_cast<decltype(m_storageOverlay)&>(m_storageOverlay)[_p] = _v; }
 
-	/// @returns true if we are in the contract-conception state and setCode is valid to call.
-	bool isFreshCode() const { return m_codeHash == c_contractConceptionCodeHash; }
+	/// @returns the hash of the account's code.
+	h256 codeHash() const { return m_codeHash; }
 
-	/// @returns true if we are either in the contract-conception state or if the account's code is not
-	/// empty.
-	bool codeBearing() const { return m_codeHash != EmptySHA3; }
-
-	/// @returns the hash of the account's code. Must only be called when isFreshCode() returns false.
-	h256 codeHash() const { assert(!isFreshCode()); return m_codeHash; }
+	bool isFreshCode() const { return m_hasNewCode; }
```

Comment:
Yes. You cannot tell now if an account is in "creation" phase. But that information should not be needed.

For the Changelog we need accounts modification to be more atomic, not packed into transnational changes. So you can not add code to any existing account -- the API will allow you to even replace the existing code with a new one... From my experience in this area, it is easier to maintain it by doing set of independent changes (e.g. create empty account, add balance, increase nonce, set code) than having API restricted transactions (e.g. create an account with this nonce, this balance, and allow setting code later).